ouroboros-consensus-cardano/src/unstable-cardano-tools/Cardano/Tools/DBAnalyser/Analysis.hs

```diff
@@ -102,47 +110,80 @@ newtype NumberOfBlocks = NumberOfBlocks { unNumberOfBlocks :: Word64 }
   deriving (Eq, Show, Num, Read)
 
 runAnalysis ::
-     forall blk .
+     forall blk.
      ( HasAnalysis blk
      , LedgerSupportsMempool.HasTxId (LedgerSupportsMempool.GenTx blk)
      , LedgerSupportsMempool.HasTxs blk
      , LedgerSupportsMempool blk
      , LedgerSupportsProtocol blk
      , LgrDbSerialiseConstraints blk
      )
-  => AnalysisName -> Analysis blk
-runAnalysis analysisName env@(AnalysisEnv { tracer }) = do
-    traceWith tracer (StartedEvent analysisName)
-    result <- go analysisName
-    traceWith tracer DoneEvent
-    pure result
+  => AnalysisName -> SomeAnalysis blk
+runAnalysis analysisName = case go analysisName of
+    SomeAnalysis p ana -> SomeAnalysis p $ \env@AnalysisEnv{ tracer } -> do
+      traceWith tracer (StartedEvent analysisName)
+      result <- ana env
```

Comment:
I'd rename `ana` to analysis. Same for `mkAna` :)

Reply:
Done :+1: